src/app/services/actions/query-action-creator-util.ts

```diff
@@ -30,28 +38,35 @@ export function anonymousRequest(dispatch: Function, query: IQuery) {
     'Content-Type': 'application/json',
   };
 
-  const options: IRequestOptions = { method: query.selectedVerb, headers};
+  const options: IRequestOptions = { method: query.selectedVerb, headers };
+
+  dispatch(queryRunningStatus({ isLoadingData: true }));
 
   return fetch(graphUrl, options)
     .then((resp) => {
+      dispatch(queryRunningStatus({ isLoadingData: false }));
       if (resp.ok) {
         return parseResponse(resp, respHeaders);
       }
       throw new Error('The request was not completed');
     })
     .then((json) =>
-
       dispatch(
         queryResponse({
           body: json,
           headers: respHeaders,
         }),
+        queryRunningStatus({ isLoadingData: false })
       ),
     )
-    .catch((error) => dispatch(queryResponseError(error)));
+    .catch((error) => dispatch(
+      queryResponseError(error),
+      queryRunningStatus({ isLoadingData: false }))
+    );
 }
 
 export function authenticatedRequest(dispatch: Function, query: IQuery) {
+  dispatch(queryRunningStatus({ isLoadingData: true }));
```

Comment:
This dispatch should happen inside the `makeRequest` function. Since a dispatch with `isLoadingData` to true should have a complementing dispatch with `isLoadingData` false. Having both complementary dispatches inside the `makeRequest` function improves readability and also prevents opportunities for errors.

Move this to line 108